src/app/components/core/user-management/update-user/update-user.component.ts

```diff
@@ -79,13 +85,37 @@ export class UpdateUserComponent implements OnInit, OnChanges {
    *  To handle role and password change method call
    */
   updateUser() {
-    if (this.showRoleSection) {
-      this.updateRole();
-    } else {
-      this.resetPassword();
+    switch (this.updateSection) {
+      case 'password':
+        this.resetPassword();
+        break;
+      case 'role':
+        this.updateRole();
+        break;
+      case 'auth':
+        this.updateAuthMethod();
+        break;
+      default:
+        break;
     }
   }
 
+  updateAuthMethod() {
+    this.userService.updateUser(this.userRecord).
+    subscribe(() => {
+        this.notify.emit();
+        this.toggleModal(false);
+        this.alertService.success('User updated successfully');
```

Comment:
what we are getting from API?

Reply:
Response is like-
`{"user_info": {"id": 2, "uname": "user", "role_id": 2, "access_method": "any", "real_name": "Normal user", "description": "normal user1"}}`

Reply:
OK. its good then